core/src/action.rs

```diff
@@ -157,8 +157,15 @@ pub enum Action {
     /// Last string is the stringified process unique id of this `hdk::get_links` call.
     GetLinks(GetLinksKey),
     GetLinksTimeout(GetLinksKey),
+    GetLinksTimeoutByTag(GetLinksKeyByTag),
+    GetLinksCount((GetLinksKey, Option<CrudStatus>)),
+    GetLinksCountByTag((GetLinksKeyByTag, Option<CrudStatus>)),
     RespondGetLinks((QueryEntryData, Vec<(Address, CrudStatus)>, String, String)),
+    RespondGetLinksCount((QueryEntryData, usize, String, String)),
+    RespondGetLinksCountByTag((QueryEntryData, usize, String)),
```

Comment:
This could have to be more comonized but having explicit enums like this tells you exactly what it is doing.